Changes to address the changed cache_root and module name in Apache 2.4.

dteirney opened the pull request:
A new mod_name variable is included in apache::mod::disk_cache as it changed from mod_disk_cache to mod_cache_disk. This is used to load the correct module and by the disk_cache.conf template.

The default cache_root changed from /var/cache/mod_proxy to /var/cache/httpd/proxy (on Redhat).

TODO: confirm if the default cache root changed on either Debian or FreeBSD.

mhaskel commented:
@dteirney thanks for the contribution, but we haven't heard from you in a while so I'm going to have to close this PR. If this is something you're still interested in working on, please address the comments and resubmit the PR.

dbeckham commented:
@dteirney This module has been broken since 1.2.0 (see https://tickets.puppetlabs.com/browse/MODULES-1410) and there are no other PRs or related fixes that I can find.  The `apache::mod::disk_cache` class does not work with Ubuntu LTS 14 and Apache 2.4. This Puppetlabs Module is broken.

I'm about to have to fix this for myself in a fork because I need to be able to use this module.  What is it that you want in order to be able to merge a PR and I'll conform to that so at least other people can use this module as well.

dteirney commented:
@mhaskel apologies for the radio silence. Been busy with work and holidays since Christmas.

@dbeckham, I've just rebased the fork at orionhealth and added in some of the last changes that were needed. I'll create a new PR for further discussion. If you are happy to have a look and perhaps add some tests that would be amazing. I'm back on holiday again for a few days so primarily pushed up some changes that I had already completed.

Possibly needs further review as I haven't looked at it in a while.

New PR at https://github.com/puppetlabs/puppetlabs-apache/pull/1037
